### ISSUE TITLE
[JavaScript] Implement new Concept Exercise: type-conversion

### ISSUE THREAD

**Issue by SleeplessByte** [exercism/v3#1892]
This issue describes how to implement the `type-conversion` concept exercise for the `javascript` track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

See the documentation _above_ (general documentation), as well as [How to implement a Concept Exercise in JavaScript](https://github.com/exercism/v3/blob/master/languages/javascript/reference/implementing-a-concept-exercise.md).

## Goal

The goal of this exercise is to teach the student how they can "cast" between types.

## JavaScript specificness

ℹ Many languages have a way to _cast_ a value into a (different) type. JavaScript does not have such constructs. However, the global objects for truthy primitives (`String`, `Number`, `Boolean`) also exist as function (`String(val)`, `Number(val)`, `Boolean(val)`) which convert the input to the _primitive_ type (non-boxed). 

## Learning objectives

- Using (some) prototype functions:
  - `* ➡ string`: `.toString()`
  - `* ➡ Array<*>`: `[Symbol.iterator]` (destructuring [works with any iterable](https://exploringjs.com/impatient-js/ch_destructuring.html#array-destructuring-works-with-any-iterable))
   - `Array<*> ➡ string`: `.join`
   - `Date ➡ number`: `.getTime`
   - `Date ➡ string`: `.toISO8601`
- Using global functions:
  - * ➡ boolean`: `Boolean(value)`
  - * ➡ number`: `Number(value)`
  - * ➡ string`: `String(value)`
  - ...* ➡ Array: `Array(...values)` 
- Using operators:
  - ` * ➡ boolean`: `!value` and `!!value`
- Using constructors:   
  - `number ➡ Date`: `new Date(unix)`
  - `string ➡ Date`: `new Date(iso8601)`

If this list can not be turned into a single exercise, you can _remove_ the `Date` related ones, and put that into a `dates` concept exercise.

## Out of scope

- `number#toString(radix)` (probably need a different exercise about non-base-10 numbers)
- `new Function(value)` (we're not teaching `eval` here, and does MORE than type conversion)
- Loose type conversion that also does invalidation:
  - `parseInt` (probably want an input validation exercise, does MORE than type conversion)
  - `parseFloat` (probably want an input validation exercise, does MORE than type conversion)
- Unary operators that basically do an implicit type cast (such as `==` does)
  - `+string` (not idiomatic)
  - `~string` (not idiomatic)
- Binary operators that do an implicit type cast (actually `==`)
  - `string ^ string` (not idiomatic)
  - `string * string` (not idiomatic)

## Concepts

- `type-conversion`

## Prerequisites

- `basics`
- ... (do you think something is missing?)

## Help

You can choose to do this solo-style, or collaborate with multiple people on this. My suggestion would be is to 

1. **first accept this issue by saying "I'd like to work on this"** (you will not get a response from me. Just go with it) and optionally request that someone works with you (and wait for a second person to accept your request).
2. Use this issue to write down what you need (collect reference material) and discuss, if necessary, what to write and what not to write.
3. Draft a PR and request the feedback from at least one other JavaScript member (I recommend just pinging the channel on slack).

**Comment by SleeplessByte** [comment]
@shubhsk88 maybe this one is interesting for you? I can also write up _new_ exercise issues if you'd like.

**Comment by shubhsk88** [comment]
@SleeplessByte  yeah but I think there is too much stuff in this concept exercise, Isn't the exercise supposed to be short .I will be happy to take that. Yeah, sure I will be happy to write a new exercise especially the `new` function and how `class`  works in javascript but hope it won't be deep for the students. What do you think?

**Comment by SleeplessByte** [comment]
It's not _that_ important to understand how `class` internally works -- but it's important how `class` makes you "think differently" when writing JavaScript. There is #1390 (#1325), no clue if they are still working on it. That said, concepts may be very complex! No problem.

> Isn't the exercise supposed to be short

Not necessarily. There mostly should just be one straightforward way to do a thing; and it should be teaching you one concept. The _length_ of the exercise isn't that much of an issue.

In this case, there is no need to implement all of the the teaching goals. If a student *understands* how to go from some type A to some type B, and understands how to search for it / look it up, that's already good enough.

**Comment by shubhsk88** [comment]
Well said @SleeplessByte  and this I will try to write this. Is this sounds good?

**Comment by SleeplessByte** [comment]
Yeah, of course!

**Comment by shubhsk88** [comment]
Can you assign me to this issue so other won't take it lol
